### DISCUSSION TITLE
RasterLayer Intrinsic/Extrinsic Parameters

### DISCUSSION THREAD

**Discussion by syberhogue** [cnr-isti-vclab/meshlab#1090] [Q&A]
Does anyone have any documentation, insight, or information on the coordinate frame of the transformation (R/T) and the intrinsics info that is needed to properly specify a RasterLayer in a .mlp ?  

_Note: I'm using this to texture a mesh using the Filter:"Parameterization + texturing from registered rasters"._

Specifically, in the .mlp file we have to define the following: (note I've placed variables here for discussion"
<VCGCamera CenterPx="cx cy" FocalMm="focal" PixelSizeMm="1 1" ViewportPx="width height " LensDistortion="k1 k2" RotationMatrix="R_4x4 " TranslationVector="T_1x4" CameraType="0"/>

### Through some trials and tribulations I've noticed the following:
- **CenterPx**, cx/cy are not taken into account correctly (possibly).  **NOTE:** If these are set to my calibrated cx/cy from calibration software, an offset is apparent if these are not exactly the 0.5*(width,height)

- **FocalMm**, the focal length of the lens/camera is usually known in pixels, so I'm using pixels here rather than Mm, don't be fooled by the designation of "millimeters".  **QUESTION**: is this the Focal length on X or Y or an average? Is it ok to use pixels as units as the case with most calibration software these days

- **PixelSizeMm** - I originally thought this might be useful to specify the aspect ratio but "1 1" is the only thing that worked.  QUESTION: can someone enlighten me as to how this is supposed to be set or the intuition behind the values here?

- **ViewportPx** is obviously the width/height of the raster image in pixels

- **LensDistortion** is 2 parameters (k1,k2) but in what units?  **QUESTION**: are these in pixels or millimeters or relative to the focal length or ..... ?

- **RotationMatrix** - a 4x4 matrix with just a rotation in it: **QUESTION**: Is this the World to Camera or Camera to World ?

- **TranslationVector** - a 1x4 vector (x,y,z,1) **QUESTION**: Is this from World to Camera or Camera to World ?

Coordinate Frame:
- what does the coordinate frame look like here?  Is it a standard OpenGL frame? Right-handed?

any insights are appreciated!

**Comment by ventusff** [comment]
I think the actual codes and code comments that defines meshlab's camera behaviors might help answer your question.
https://github.com/cnr-isti-vclab/vcglib/blob/d1d106a4ce3a46d42fdfa665bb3e17b0c6ba61bd/vcg/math/camera.h
https://github.com/cnr-isti-vclab/vcglib/blob/d1d106a4ce3a46d42fdfa665bb3e17b0c6ba61bd/vcg/math/shot.h
